base_layer/core/src/covenants/output_set.rs

```diff
@@ -51,6 +51,8 @@ impl<'a> OutputSet<'a> {
         *self = new_set;
     }
 
+    /// Given a functional that returns a bool on a transaction output,
+    /// it filters the underlying set of `TransactionOutput`s.
```

Comment:
```suggestion
    /// Retains all outputs for which the given predicate f return true.
```
I should probably have named 'f' as 'predicate'